subsys/net/ip/net_if.c

```diff
@@ -2192,6 +2192,24 @@ void net_if_init(void)
 #endif
 	}
 #endif /* CONFIG_NET_IPV6 */
+
+#if defined(CONFIG_NET_VLAN)
+	/* Make sure that we do not have too many network interfaces
+	 * compared to the number of VLAN interfaces.
+	 */
+	for (iface = __net_if_start, if_count = 0;
+	     iface != __net_if_end; iface++) {
+		if (net_if_l2(iface) == &NET_L2_GET_NAME(ETHERNET)) {
+			if_count++;
+		}
+	}
+
+	if (if_count > CONFIG_NET_VLAN_COUNT) {
+		NET_WARN("You have configured only %d VLAN interfaces"
+			 " but you have %d network interfaces.",
+			 CONFIG_NET_VLAN_COUNT, if_count);
```

Comment:
What about configuring vlan at runtime? (Is it possible and/or a planned feature ?)

Reply:
VLAN is created statically but enabled / configured at runtime. Because of this it is possible that user made an error in prj.conf file, for example not using ETH_NET_DEVICE_INIT() when creating network interface but still enabling VLAN. This can lead invalid setup. In order to avoid weird errors, I added a check about this scenario so that user is able to get some feedback about the issue.

Reply:
Ok